src/pybind/mgr/dashboard/services/auth/oauth2.py

```diff
@@ -23,17 +22,14 @@ class OAuth2Config(BaseAuth.Config):
     def enabled():
         return mgr.get_module_option('sso_oauth2')
 
-    def to_dict(self) -> 'BaseAuth.Config':
-        return self.OAuth2Config()
+    @staticmethod
+    def to_dict() -> 'BaseAuth.Config':
+        return OAuth2.OAuth2Config()
```

Comment:
Same again:
```suggestion
    def to_dict(self) -> 'BaseAuth.Config':
        return self.OAuth2Config(self)
```
What I don't really get is how are we dumping as dict the state of the current object if we're returning a generic dict?

Reply:
Updated it. I think it is properly typed out now..